homeassistant/components/legrandinone.py

```diff
@@ -0,0 +1,473 @@
+    iobl_type = None
+    iobl_unit = None
+    _protocol = None
+
+    def __init__(self, device_id, protocol, initial_event=None, name=None,
+                 iobl_media='plc', iobl_comm_mode='unicast'):
+        """Initialize the device."""
+        self._initial_event = initial_event
+        self.legrand_id = device_id
+        self.iobl_mode = iobl_comm_mode
+        self.iobl_media = iobl_media
+        self._protocol = protocol
+
+        if name:
+            self._name = name
+        else:
+            self._name = device_id
+
+    @callback
+    def handle_event_callback(self, event):
+        """Handle incoming event for device type."""
+        # Call platform specific event handler
+        self._handle_event(event)
+
+        # Propagate changes through ha
+        self.async_schedule_update_ha_state()
+
+    def _handle_event(self, event):
+        """Platform specific event handler."""
+        raise NotImplementedError()
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def name(self):
+        """Return a name for the device."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        if self.assumed_state:
+            return False
+        return self._state
+
+    @property
+    def assumed_state(self):
+        """Assume device state until first device event sets state."""
+        return self._state is None
+
+    def available(self):
+        """Return True if entity is available."""
+        return self._available
+
+    @callback
+    def _availability_callback(self, availability, protocol):
+        """Update availability state."""
+        self._available = availability
+        self._protocol = protocol
+        self.async_schedule_update_ha_state()
+
+    async def async_added_to_hass(self):
+        """Register update callback."""
+        # Register id and aliases
+        self.hass.data[DATA_ENTITY_LOOKUP][
+            EVENT_KEY_COMMAND][self.legrand_id].append(self.legrand_id)
+
+        async_dispatcher_connect(self.hass, SIGNAL_AVAILABILITY,
+                                 self._availability_callback)
+        async_dispatcher_connect(self.hass,
+                                 SIGNAL_HANDLE_EVENT.format(self.legrand_id),
+                                 self.handle_event_callback)
+
+        # Process the initial event now that the entity is created
+        if self._initial_event:
+            self.handle_event_callback(self._initial_event)
+
+
+class LegrandInOneCommand(LegrandInOneDevice):
+    """Class to make IOBL command interface available to entities.
+
+    This class is to be inherited by every Entity class that is actionable
+    (switches/lights). It exposes the IOBL command interface for these
+    entities.
+
+    The IOBL interface is managed as a class level and set during setup (and
+    reset on reconnect).
+    """
+
+    def is_connected(self):
```

Comment:
Why do we need this when we have the `available` property?

Reply:
Removed